src/Data/List/Properties.agda

```diff
@@ -624,6 +626,47 @@ length-replicate : ∀ n {x : A} → length (replicate n x) ≡ n
 length-replicate zero    = refl
 length-replicate (suc n) = cong suc (length-replicate n)
 
+lookup-replicate : ∀ n (x : A) (i : Fin (length (replicate n x))) → lookup (replicate n x) i ≡ x
```

Comment:
Consider `let` binding the duplicated expression `xs = replicate n x` for the sake of legibility:
```agda
lookup-replicate : ∀ n (x : A) → let xs = replicate n x in (i : Fin (length xs)) → lookup xs i ≡ x
```
(here, and in `CHANGELOG`)

Reply:
Better, however: you know that `length (replicate n x)` is equal to `n`, so no need to consider `i : Fin (length (replicate n x))` when `i : Fin n` will suffice?

Reply:
Replacing `i : Fin (length (replicate n x))` with `i: Fin n` gives me this -
```
n != (foldr (const suc) 0 (replicate n x)) of type ℕ
when checking that the expression i has type
Fin (length (replicate n x))
```

Reply:
I think you'd need
```agda
lookup-replicate : ∀ n (x : A) (i : Fin n) → lookup (replicate n x) (cast (sym (length-replicate n x)) i) ≡ x
```
which does not really feel like a large improvement?

[The problem really stems from the type of `lookup` which has a computation in it that doesn't reduce, and infects everything else.]

Reply:
Thanks! I'll update this and `lookup-iterate`.